superset/db_engine_specs.py

```diff
@@ -675,18 +653,20 @@ def handle_cursor(cls, cursor, query, session):
             hive.ttypes.TOperationState.RUNNING_STATE,
         )
         polled = cursor.poll()
+        import logging
         while polled.operationState in unfinished_states:
-            query = session.query(type(query)).filter_by(id=query.id)
+            query = session.query(type(query)).filter_by(id=query.id).one()
             if query.status == QueryStatus.STOPPED:
                 cursor.cancel()
                 break
-
             resp = cursor.fetch_logs()
-            if resp and resp.log:
-                progress = cls.progress(resp.log)
+            if resp and len(resp) > 0:
```

Comment:
AFAIU __len__ is called if __nonzero__ is not defined so should be covered too
https://docs.python.org/2/reference/datamodel.html#object.__nonzero__